daemon/emer-aggregate-tally.c

```diff
@@ -0,0 +1,373 @@
+
+G_DEFINE_TYPE (EmerAggregateTally, emer_aggregate_tally, G_TYPE_OBJECT)
+
+enum {
+  PROP_0,
+  PROP_PERSISTENT_CACHE_DIRECTORY,
+  N_PROPS,
+};
+
+static GParamSpec *properties[N_PROPS] = { NULL, };
+
+static inline GVariant *
+aggregate_timer_to_variant (guint32    unix_user_id,
+                            GVariant  *event_id,
+                            GVariant  *aggregate_key,
+                            GVariant  *payload,
+                            guint32    counter,
+                            gint64     monotonic_time_us)
+{
+  return g_variant_new ("(vvuumv)",
+                        event_id,
+                        aggregate_key,
+                        unix_user_id,
+                        counter,
+                        payload);
+}
+
+static gchar *
+get_tally_path_from_aggregate_timer (EmerAggregateTally *self,
+                                     guint32             unix_user_id,
+                                     GVariant           *event_id,
+                                     GVariant           *aggregate_key,
+                                     const char         *date)
+{
+  g_autofree gchar *aggregate_key_str = NULL;
+  g_autofree gchar *event_id_str = NULL;
+  g_autofree gchar *filename = NULL;
+
+  aggregate_key_str = g_variant_print (aggregate_key, FALSE);
+  event_id_str = g_variant_print (event_id, FALSE);
+  filename = g_strdup_printf ("%s-%u-%s",
+                              aggregate_key_str,
+                              unix_user_id,
+                              event_id_str);
+
+  return g_build_filename (self->persistent_cache_directory,
+                           "aggregate-timers",
+                           date,
+                           filename,
+                           NULL);
+}
+
+static guint32
+find_previous_aggregate_timer_counter (EmerAggregateTally *self,
+                                       guint32             unix_user_id,
+                                       GVariant           *event_id,
+                                       GVariant           *aggregate_key,
+                                       const char         *date)
+{
+  g_autofree gchar *tally_path = NULL;
+  g_autofree gchar *contents = NULL;
+  g_autoptr(GVariant) variant = NULL;
+  g_autoptr(GVariant) unused_event_id = NULL;
+  g_autoptr(GVariant) unused_aggregate_key = NULL;
+  g_autoptr(GVariant) unused_payload = NULL;
+  g_autoptr(GError) error = NULL;
+  guint32 previous_counter;
+  guint32 unused_unix_user_id;
+  gsize size;
+
+  tally_path = get_tally_path_from_aggregate_timer (self,
+                                                    unix_user_id,
+                                                    event_id,
+                                                    aggregate_key,
+                                                    date);
+
+  if (!g_file_test (tally_path, G_FILE_TEST_EXISTS))
+    return 0;
+
+  g_file_get_contents (tally_path, &contents, &size, &error);
+  if (error)
```

Comment:
Personally I prefer this style, but I accept it is a matter of preference :)

```suggestion
  if (!g_file_get_contents (tally_path, &contents, &size, &error))
```